teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -112,9 +112,186 @@ be handled, or user scenarios that will be affected and must be accomodated.
+
+_Note that, `spec.pipelineSpec.tasks.taskSpec.spec` is holding the custom task spec._
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: pr-loop-example
+spec:
+  pipelineSpec:
+    tasks:
+      - name: first-task
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              imagePullPolicy: IfNotPresent
+              script: |
+                #!/usr/bin/env bash
+                echo "I am the first task before the loop task"
+      - name: loop-task
+        runAfter:
+          - first-task
+        params:
+          - name: message
+            value:
+              - I am the first one
+              - I am the second one
+              - I am the third one
+        taskSpec:
+          apiVersion: custom.tekton.dev/v1alpha1
+          kind: PipelineLoop
+          spec:
+            iterateParam: message
+            pipelineSpec:
+              params:
+                - name: message
+                  type: string
+              tasks:
+                - name: echo-loop-task
+                  params:
+                    - name: message
+                      value: $(params.message)
+                  taskSpec:
+                    params:
+                      - name: message
+                        type: string
+                    steps:
+                      - name: echo
+                        image: ubuntu
+                        imagePullPolicy: IfNotPresent
+                        script: |
+                          #!/usr/bin/env bash
+                          echo "$(params.message)"
+      - name: last-task
+        runAfter:
+          - loop-task
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              imagePullPolicy: IfNotPresent
+              script: |
+                #!/usr/bin/env bash
+                echo "I am the last task after the loop task"
+
+```
+
+`Tektoncd/pipeline` can only validate the structure and fields it knows about, validation of the
+custom task spec field(s) is delegated to the custom task controller.
+
+A custom controller may still choose to not support one of `Spec` or `Ref` based 
+`Run` or `PipelineRun` specification. This can be done by implementing validations at the custom controller end.
+If the custom controller did not respond in any of the ways i.e. either validation errors or reconcile CRD,
+then, a `PipelineRun` or a `Run` will wait until the timeout and mark the status as `Failed`.
+
+What is the fate of an existing custom controller developed prior to the implementation of this TEP. If the 
+custom controller implemented a validation for missing a `Ref`, then the `PipelineRun` or `Run` 
+missing a `Ref` will fail immediately with configured error and if however, no validation was
+implemented for missing a `Ref`, then it can even lead to `nil dereference errors` or have the same fate
```

Comment:
thank you for documenting this here. We highly encourage the custom controller to have some kind of validation on either one of the fields being initialized `ref` or `spec`.